### DISCUSSION TITLE
Connecting to locally hosted APIs from Desktop app

### DISCUSSION THREAD

**Discussion by jase52476** [hoppscotch/hoppscotch#3928] [Q&A] [answered]
I have searched both the documentation and the discussions, and I cannot figure out how to connect to localhost (for unit testing/troubleshooting/debugging locally) via the **desktop** app.

Does the desktop app support this feature? I installed the browser extension, but it does not work for me while using the desktop app.

I have created a fake URI like localhost.mydomain.com, and put it into my hosts file, and I even have a digicert with localhost.mydomain.com in the servers list, but Hoppscotch will not allow me to make a call to my local host alias.

I read this [discussion](https://github.com/hoppscotch/hoppscotch/discussions/2051), but it does not seem to work for the desktop app.

Any help is appreciated.

Thank you.

**Comment by SanskritiHarmukh** [comment]
Hi @jase52476 , did you try switching to "Native" interceptor at desktop app while making requests for localhost endpoints?

**Reply by SanskritiHarmukh** [reply]
You can find the **Native interceptor** at the left bottom with a "security" icon.
![Screenshot from 2024-03-23 03-02-32](https://github.com/hoppscotch/hoppscotch/assets/74777863/0bd4e506-8a10-4bec-9c8a-41e6a62f8e49)

**Reply by jase52476** [reply]
Yes tried both. Tried putting my machine name with actual domain, and it didn't work either. 

I put this into my hosts file : 
127.0.0.1 : localhost.mydomain.com

It's saying unable to reach API endpoint. I'm able to hit it fine in PostMan using the same uri, so I know the API works.

**Comment by AndrewBastin** [comment]
Hi @jase52476 ,

I tried it myself locally and it does work. So I cannot reproduce it from my end. Is there anything else notable you can add as more info ?

<img width="1727" alt="image" src="https://github.com/hoppscotch/hoppscotch/assets/9131943/e3a501e4-9924-43dc-910a-b74befd2c3ef">

**Reply by tinycocha** [reply]
please try local endpoint, eg. https://ip+port

**Reply by jase52476** [reply]
I'm using https and the port is 443, so maybe that has something to do with it?

**Reply by jase52476** [reply]
I finally got it to hit my server, so looks like it's with the https:

![image](https://github.com/hoppscotch/hoppscotch/assets/18239060/3201212d-5bb4-470a-a994-f63faf7809fd)

Let me try fixing my cert b/c the localhost.mydomain.com cert I had is expired.  Postman doesn't care, it will just put an error message saying the certificate is invalid or expired, but looks like Hoppscotch can't ignore bad cert?

**Reply by jase52476** [reply, marked as answer]
Got it working.  Need to point localhost to a URL like localhost.mydomain.com and make sure that you have a certificate that has the same domain in the certificate.

**Reply by adaspender** [reply]
Had a similar issue resulting in a could not send request error. Interceptor Native, everything seemed correct. Calls only started working after setting Verify SSL Certifactes to off in the Settings.

![image](https://github.com/user-attachments/assets/8120b45e-eccc-4dfe-ba0b-19c92f58cfcd)
![image](https://github.com/user-attachments/assets/07c72809-bfbd-4d7a-ac2a-56fad81245d8)

**Reply by mycarrysun** [reply]
Why don't I have that option to try `Native`?

![image](https://github.com/user-attachments/assets/37be33ae-825f-446c-a284-825ee0e29fe7)

**Comment by LadyNay** [comment]
Same issue on my end with AIO Docker container
